airgen/src/lib.rs

```diff
@@ -0,0 +1,193 @@
+//! Compilation from powdr machines to AIRs
+
```

Comment:
Was this file just moved from somewhere or are there new things here?

Reply:
TL;DR: `airgen` is the extraction of the part of `asm_to_pil` which is not ASM specific. So it's `asm_to_pil` moved here and stripped down.

Before:
`asm_to_pil` turns each machine into an AIR, reducing the ASM to constraints.

After:
`asm_to_pil` only turns each machine into a another machine where all the ASM stuff (rom, instructions*, registers) is turned into constraints.

`airgen` turns each machine into an AIR

The reasoning is that after `asm_to_pil`, all machines are block machines, so we can do more processing steps before hitting `airgen`

*external instructions are still kept, so we still have some ASM information after `asm_to_pil`. This is fixed in #464 where external instructions are turned into links, which are the low-level type of machine-to-machine interaction.

Reply:
I see. Should `asm_to_pil` be renamed then? Or is the output still an `Analyzed` PIL?

Reply:
The output is the same as the input. Maybe `asm_encoder` ?